primitives/xcm/Cargo.toml

```diff
@@ -8,6 +8,7 @@ log = { workspace = true, default-features = false }
 parity-scale-codec = { workspace = true }
 frame-support = { workspace = true }
 frame-system = { workspace = true }
+runtime-common = { workspace = true }
```

Comment:
A primitive should not rely on the actual runtime. Otherwise there will be an inter-dependency. To alleviate this the adapter can take a generic type and probably add some constraints.